### DISCUSSION TITLE
Be able to customize file preview url, name display, click url via function

### DISCUSSION THREAD

**Discussion by adamberecz** [vueform/vueform#239] [Ideas]
Potentially also allow object as stage 3 file value.

Related discussion:
https://github.com/vueform/vueform/discussions/199

**Comment by aa-ndrej** [comment]
> So just to make sure I understand it correctly: you have uploaded and persisted a file ([stage 3](https://vueform.com/docs/file-uploads#image-upload-diagram)) and the `FileElement`'s value is eg. `blue-orange.png` or `https://domain.com/images/blue-orange.png` and instead of `blue-orange.png` you want to show something like `Blue Orange` for the file?
>
> <img alt="Screenshot 2024-04-08 at 17 43 11" width="228" src="https://private-user-images.githubusercontent.com/16024069/320555424-02c85d71-b3f7-4b68-af7d-2e13a0c33528.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg1MTkyMTUsIm5iZiI6MTc2ODUxODkxNSwicGF0aCI6Ii8xNjAyNDA2OS8zMjA1NTU0MjQtMDJjODVkNzEtYjNmNy00YjY4LWFmN2QtMmUxM2EwYzMzNTI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE1VDIzMTUxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMxNTZlYWFjMGVkMjkyMGZmYWI1NDVlMGQ5NjIxM2YzMWY4MTc2Mzg4MGUwMmVhYWFjMjMyNTgwZThhY2ZjMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uQRkQIAWzzhYEW8o5D1kZEIWNNCrR1KXj5ZnK9nPX8Q">

@adamberecz is this somehow possible as of today?
Our URLs for persisted files contain file IDs while we want to show the original file name that comes from whithin our database. 🤔

**Reply by adamberecz** [reply]
The API haven't changed since so everything written there stands correct.
